tests/upgrade_path/upgrade_helpers.py

```diff
@@ -117,3 +118,60 @@ def check_reboot_cause(duthost, expected_cause):
     reboot_cause = get_reboot_cause(duthost)
     logging.info("Checking cause from dut {} to expected {}".format(reboot_cause, expected_cause))
     return reboot_cause == expected_cause
+
+
+def copy_sonic_image_into_onie(image_path, localhost, dut_ip):
+    """
+    Copy SONiC image into ONIE
+    """
+
+    dst_path = "/tmp/sonic.bin"
+    if image_path.startswith("http"):
+        localhost.shell(
+            "sshpass -v ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@{} 'wget -O {} {}'".format(
+                dut_ip, dst_path, image_path))
+    else:
+        localhost.shell(
+            "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null {} root@{}:{}".format(image_path,
+                                                                                                   dut_ip, dst_path))
+    return dst_path
+
+
+def install_base_sonic_image(duthost, localhost, image_path, tbinfo, local_mg_path, downgrade_type):
+    """
+    Install base SONiC image
+    """
+    if downgrade_type == "sonic":
+        target_version = install_sonic(duthost, image_path, tbinfo)
+        # Perform a cold reboot
+        logger.info("Cold reboot the DUT to make the base image as current")
+        reboot(duthost, localhost)
+        check_sonic_version(duthost, target_version)
```

Comment:
would be nice to have different methods for deploy from sonic and deploy from onie